Application/Activities/UpdateAttendance.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Application.Interfaces;
+using Domain.Enitities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Persistence.Data;
+
+namespace Application.Activities
+{
+    public class UpdateAttendance
+    {
+        public class Command : IRequest<Unit>
+        {
+            public Guid Id { get; set; }
+        }
+        public class Handler : IRequestHandler<Command, Unit>
+        {
+            private readonly DataContext _dataContext;
+            private readonly IUserAccesor _userAccesor;
+
+            public Handler(DataContext dataContext, IUserAccesor userAccesor)
+            {
+                _dataContext = dataContext;
+                _userAccesor = userAccesor;
+            }
+
+            public async Task<Unit> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var activity = await _dataContext.Activities.Include(a => a.Attendees)
+                    .ThenInclude(u => u.AppUser)
+                    .SingleOrDefaultAsync(x => x.Id == request.Id);
+
+                if( activity == null) return Unit.Value; ;
+
+                var user = await _dataContext.Users.FirstOrDefaultAsync(x => x.UserName == _userAccesor.GetUserName());
+
+                if (user == null) return Unit.Value;
+
+                var hostUserName = activity.Attendees.FirstOrDefault(x => x.IsHost)?.AppUser?.UserName;
+                var attendance = activity.Attendees.FirstOrDefault(x => x.AppUser.UserName == user.UserName);
+
+                if (attendance != null && hostUserName == user.UserName)
+                {
+                    activity.IsCancelled = !activity.IsCancelled;
+                } 
+
+                if(attendance != null && hostUserName != user.UserName)
```

Comment:
Format this file because there are some empty spaces that are missing between some characters

Reply:
done.